src/logic/nep330_build/output.rs

```diff
@@ -59,9 +67,48 @@ mod rust_legacy {
             parent: &tracing::Span::none(),
             "assumed artifact result path for a rust crate docker build: `{}`", path
         );
+
+        common::path_sane_check(&path, "rust crate")?;
+
+        Ok(path)
+    }
+}
+
+pub mod explicit_metadata {
+    use eyre::ContextCompat;
+
+    use crate::logic::NEP330_REPO_MOUNT;
+
+    use super::common;
+
+    pub fn wasm_output_path(
+        output_wasm_path: &str,
+        contract_source_workdir: camino::Utf8PathBuf,
+    ) -> eyre::Result<camino::Utf8PathBuf> {
+        let base = camino::Utf8PathBuf::from(NEP330_REPO_MOUNT);
+        let subpath = camino::Utf8PathBuf::from(output_wasm_path);
+
+        let relative_path = pathdiff::diff_utf8_paths(&subpath, &base).wrap_err(format!(
+            "cannot compute contract output pathdiff from mount point {}",
+            NEP330_REPO_MOUNT
+        ))?;
+        let wasm_path = contract_source_workdir.join(relative_path);
```

Comment:
I'm not sure I understand why we're trying to convert the absolute path to a relative path to /home/near/code, and then apply that relative path to contract_source_workdir.

It seems to me that, in the end, we’d end up with the same path anyway...

So I'm not sure what exactly we're trying to check here.

Reply:
As per internal discussion, this apply the relative path inside the container to the system path outside container.

e.g., 

contract build in the os - /home/near/code/target/near/contract.wasm
your system path is -> /home/akorchyn/work/near/contract

this will take e.g. ./target/near/contract.wasm and apply to the your system path,